### ISSUE TITLE
Move mraa_i2c_mode_t from types.h to i2c.h

### ISSUE THREAD

**Issue by eschaton** [intel-iot-devkit/mraa#157]
The header `<mraa/types.h>` purports to be for basic shared types, but it contains an i2c-specific type, `mraa_i2c_mode_t`. That should really be in `<mraa/i2c.h>` instead.

**Comment by arfoll** [comment]
Agreed, the struct should 'ideally' be duplicated in the C++ header and the C header so that SWIG uses the correct name. The reason for throwing it in the types.h is that swig takes that file (but not i2c.h) and will strip the prefix away (since we have the namespace/module/package name). So it really depends on whether you prefer enum duplication or things in the right place :) I'm happy to take a PR to change this.

**Comment by eschaton** [comment]
I can't submit a PR, sorry.
